tests/pr_test.go

```diff
@@ -71,3 +80,110 @@ func TestRunFSCloudExample(t *testing.T) {
+
+	prefix := fmt.Sprintf("vsi-slz-%s", strings.ToLower(random.UniqueId()))
+	realTerraformDir := "./resources"
+	tempTerraformDir, _ := files.CopyTerraformFolderToTemp(realTerraformDir, fmt.Sprintf(prefix+"-%s", strings.ToLower(random.UniqueId())))
+	tags := common.GetTagsFromTravis()
+
+	// Verify ibmcloud_api_key variable is set
+	checkVariable := "TF_VAR_ibmcloud_api_key"
+	val, present := os.LookupEnv(checkVariable)
+	require.True(t, present, checkVariable+" environment variable not set")
+	require.NotEqual(t, "", val, checkVariable+" environment variable is empty")
+
+	// Programmatically determine region to use based on availability
+	region, _ := testhelper.GetBestVpcRegion(val, "../common-dev-assets/common-go-assets/cloudinfo-region-vpc-gen2-prefs.yaml", "eu-de")
+
+	logger.Log(t, "Tempdir: ", tempTerraformDir)
+	existingTerraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: tempTerraformDir,
+		Vars: map[string]interface{}{
+			"prefix":        prefix,
+			"region":        region,
+			"resource_tags": tags,
+		},
+		// Set Upgrade to true to ensure latest version of providers and modules are used by terratest.
+		// This is the same as setting the -upgrade=true flag with terraform.
+		Upgrade: true,
+	})
+
+	terraform.WorkspaceSelectOrNew(t, existingTerraformOptions, prefix)
+	_, existErr := terraform.InitAndApplyE(t, existingTerraformOptions)
+	if existErr != nil {
+		assert.True(t, existErr == nil, "Init and Apply of temp existing resource failed")
+	} else {
+		outputJson := terraform.OutputJson(t, existingTerraformOptions, "vpc_data")
+		var managementVpcID string
+		var vpcs []struct {
+			VpcID   string `json:"vpc_id"`
+			VpcName string `json:"vpc_name"`
+		}
+		// Unmarshal the JSON data into the struct
+		if err := json.Unmarshal([]byte(outputJson), &vpcs); err != nil {
+			fmt.Println(err)
+			return
+		}
+		// Loop through the vpcs and find the vpc_id when vpc_name is "<prefix>-management-vpc"
+		for _, vpc := range vpcs {
+			if vpc.VpcName == fmt.Sprintf("%s-management-vpc", prefix) {
+				managementVpcID = vpc.VpcID
+			}
+		}
+		// ------------------------------------------------------------------------------------
+		// Deploy landing-zone extension
+		// ------------------------------------------------------------------------------------
+		options := testhelper.TestOptionsDefault(&testhelper.TestOptions{
+			Testing:      t,
+			TerraformDir: "extension/landing-zone",
+			// Do not hard fail the test if the implicit destroy steps fail to allow a full destroy of resource to occur
+			ImplicitRequired: false,
+			TerraformVars: map[string]interface{}{
+				"prefix":                     prefix,
+				"region":                     region,
+				"resource_group":             fmt.Sprintf("%s-management-rg", prefix),
+				"existing_kms_instance_guid": permanentResources["hpcs_south"],
+				"boot_volume_encryption_key": permanentResources["hpcs_south_root_key_crn"],
```

Comment:
Can we use the key protect provisioned by SLZ instead? I think we will end up getting clashing auth policies if we use HPCS

Reply:
@ocofaigh, The crn details of the generated kms and the keys and not outputted in any of the patterns. Will have to make changes in the SLZ to output the key details first.

Reply:
@Aashiq-J This is coming in https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/pull/583